src/controllers/apps/video-app/comment.controller.js

```diff
@@ -0,0 +1,256 @@
+import mongoose, { isValidObjectId } from "mongoose";
+import { Comment } from "../../../models/apps/video-app/comment.model.js";
+import { ApiError } from "../../../utils/ApiError.js";
+import { ApiResponse } from "../../../utils/ApiResponse.js";
+import { asyncHandler } from "../../../utils/asyncHandler.js";
+import { Video } from "../../../models/apps/video-app/video.model.js";
+import { getMongoosePaginationOptions } from "../../../utils/helpers.js";
+
+/**
+ * Retrieves the comments for a specific video.
+ *
+ * @param {Object} req - The request object.
+ * @param {Object} res - The response object.
+ * @returns {Object} - The response object containing the video comments.
+ * @throws {ApiError} - If the video comments are not found.
+ */
+const getVideoComments = asyncHandler(async (req, res) => {
+  try {
+    const { videoId } = req.params;
+
+    if (!isValidObjectId(videoId)) {
+      throw new ApiError(400, "Invalid Id");
+    }
+
+    const { page = 1, limit = 10 } = req.query;
+
+    const videosCommentAggregate = Video.aggregate([
```

Comment:
Why are we aggregating on `Video` model while fetching comments? We should query the `Comment` model